### ISSUE TITLE
Support nondefault kernel

### ISSUE THREAD

**Issue by fecet** [untitled-ai/jupyter_ascending#49]
Current setup need us to install jupyter_asceding for each virtual environment, which means we have to install jupyter notebook for each envs per #25  

Maybe a cleaner and more elegant way to do this is install Jupyter and all plugins in a base envs then we can choose different envs as different kernels from it.

**Comment by zplizzi** [comment]
I don't think this is possible - there's `jupyter_ascending` code running in the kernel, so I'd imagine it has to be installed in each. Annoyingly the docs on jupyter's extension system seem to have disappeared so it's hard to confirm how this all works.

**Comment by fecet** [comment]
Open different notebook for different envs would be painful as we have to set different port every time if we cannot have such features.

**Comment by zplizzi** [comment]
I'm not sure I understand. You should be able to have one instance of jupyter running, which can have multiple notebooks with multiple kernels, as long as the environment for each kernel has `jupyter_ascending` installed. This won't require multiple ports. One component of `jupyter_ascending` runs in the main `jupyter` process, receives commands from the user's editor, and forwards those commands to each of potentially multiple kernels - which each themselves are running code that listens and executes those commands.

**Comment by fecet** [comment]
It would be good enough If they can live in one instance of Jupyter. But install jupyter_asceding for a non-default env still complain cannot find a paired notebook. Should I install jupyter for other envs?
